Adapters/Vungle/ISVungleAdapter/ISVungleBannerAdapterRouter.m

```diff
@@ -0,0 +1,195 @@
+//
+//  ISVungleBannerAdapterRouter.m
+//  ISVungleAdapter
+//
+//  Copyright © 2020 ironSource. All rights reserved.
+//
+
+#import "ISVungleBannerAdapterRouter.h"
+#import "ISVungleAdapter.h"
+
+@interface ISVungleBannerAdapterRouter ()
+
+@property (nonatomic, strong, nullable) NSString *bidPayload;
+
+@end
+
+@implementation ISVungleBannerAdapterRouter
+
+- (instancetype)initWithPlacementID:(NSString *)placementID
+                           delegate:(id<ISBannerAdapterDelegate>)delegate {
+    if (self = [super init]) {
+        _placementID = placementID;
+        _delegate = delegate;
+        _bannerState = UNKNOWN;
+        _bidPayload = nil;
+        _bannerSize = BannerSizeRegular;
+    }
+
+    return self;
+}
+
+- (void)loadBannerAd {
+    self.bannerAd = [[VungleBanner alloc] initWithPlacementId:self.placementID size:self.bannerSize];
+    self.bannerAd.delegate = self;
+    dispatch_async(dispatch_get_main_queue(), ^{
+        self.adView = [[UIView alloc] initWithFrame:[self getAdViewRect:self.bannerSize]];
+    });
+
+    // Because there's no auto-caching in 7.0 VungleAds,the Ad would
+    // never be ready until a load call is made. We don't need to
+    // check [VungleBanner canPlayAd] here.
+    [self.bannerAd load:self.bidPayload];
+}
+
+- (void)showBannerAd {
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self.bannerAd presentOn:self.adView];
+    });
+    self.bannerState = SHOWING;
+}
+
+- (void)setBidPayload:(NSString * _Nullable)bidPayload {
+    _bidPayload = bidPayload;
+}
+
+- (void)setSize:(ISBannerSize *)size {
+    self.bannerSize = [self getBannerSize:size];
+}
+
+- (void)destroy {
+    [self.adView removeFromSuperview];
+    self.adView = nil;
+    self.bannerAd.delegate = nil;
+    self.bannerAd = nil;
+}
+
+- (void)bannerAdInitSuccess {
+    [self.delegate adapterBannerInitSuccess];
+}
+
+- (void)bannerAdInitFailed:(NSError *)error {
+    [self.delegate adapterBannerInitFailedWithError:error];
+}
+
+#pragma mark - Helper
+
+- (BannerSize)getBannerSize:(ISBannerSize *)size {
+    BannerSize vungleAdSize = BannerSizeRegular;
+
+    if ([size.sizeDescription isEqualToString:@"RECTANGLE"]) {
+        vungleAdSize = BannerSizeMrec;
+    } else if ([size.sizeDescription isEqualToString:@"SMART"]) {
+        if (UI_USER_INTERFACE_IDIOM() == UIUserInterfaceIdiomPad) {
```

Comment:
UI_USER_INTERFACE_IDIOM() is depricated. please update. @YueVungle

Reply:
@ManojBudumuru-vungle Thanks a lot for the guidance, updated. ：）